rpc-narpc/src/main/java/com/ibm/crail/namenode/rpc/tcp/TcpNameNodeResponse.java

```diff
@@ -0,0 +1,293 @@
+import com.ibm.crail.utils.CrailUtils;
+import com.ibm.narpc.NaRPCMessage;
+
+public class TcpNameNodeResponse extends RpcResponseMessage implements RpcNameNodeState, NaRPCMessage {
+	public static final Logger LOG = CrailUtils.getLogger();
+	public static final int CSIZE = 2*Short.BYTES + Math.max(RpcResponseMessage.GetBlockRes.CSIZE, RpcResponseMessage.RenameRes.CSIZE);
+	
+	private short type;
+	private short error;
+	private RpcResponseMessage.VoidRes voidRes;
+	private RpcResponseMessage.CreateFileRes createFileRes;
+	private RpcResponseMessage.GetFileRes getFileRes;
+	private RpcResponseMessage.DeleteFileRes delFileRes;
+	private RpcResponseMessage.RenameRes renameRes;
+	private RpcResponseMessage.GetBlockRes getBlockRes;
+	private RpcResponseMessage.GetLocationRes getLocationRes;	
+	private RpcResponseMessage.GetDataNodeRes getDataNodeRes;
+	private RpcResponseMessage.PingNameNodeRes pingNameNodeRes;
+	
+	public TcpNameNodeResponse() {
+		this.type = 0;
+		this.error = 0;
+		this.voidRes = new RpcResponseMessage.VoidRes();
+		this.createFileRes = new RpcResponseMessage.CreateFileRes();
+		this.getFileRes = new RpcResponseMessage.GetFileRes();
+		this.delFileRes = new RpcResponseMessage.DeleteFileRes();
+		this.renameRes = new RpcResponseMessage.RenameRes();
+		this.getBlockRes = new RpcResponseMessage.GetBlockRes();
+		this.getLocationRes = new RpcResponseMessage.GetLocationRes();
+		this.getDataNodeRes = new RpcResponseMessage.GetDataNodeRes();
+		this.pingNameNodeRes = new RpcResponseMessage.PingNameNodeRes();
+	}
+	
+	public TcpNameNodeResponse(RpcResponseMessage.VoidRes message) {
+		this.type = message.getType();
+		this.voidRes = message;
+	}
+	
+	public TcpNameNodeResponse(RpcResponseMessage.CreateFileRes message) {
+		this.type = message.getType();
+		this.createFileRes = message;
+	}	
+	
+	public TcpNameNodeResponse(RpcResponseMessage.GetFileRes message) {
+		this.type = message.getType();
+		this.getFileRes = message;
+	}
+	
+	public TcpNameNodeResponse(RpcResponseMessage.DeleteFileRes message) {
+		this.type = message.getType();
+		this.delFileRes = message;
+	}	
+	
+	public TcpNameNodeResponse(RpcResponseMessage.RenameRes message) {
+		this.type = message.getType();
+		this.renameRes = message;
+	}
+	
+	public TcpNameNodeResponse(RpcResponseMessage.GetBlockRes message) {
+		this.type = message.getType();
+		this.getBlockRes = message;
+	}
+	
+	public TcpNameNodeResponse(RpcResponseMessage.GetLocationRes message) {
+		this.type = message.getType();
+		this.getLocationRes = message;
+	}
+	
+	public TcpNameNodeResponse(RpcResponseMessage.GetDataNodeRes message) {
+		this.type = message.getType();
+		this.getDataNodeRes = message;
+	}	
+	
+	public TcpNameNodeResponse(RpcResponseMessage.PingNameNodeRes message) {
+		this.type = message.getType();
+		this.pingNameNodeRes = message;
+	}
+	
+	public void setType(short type) throws Exception {
+		this.type = type;
+		switch(type){
+		case RpcProtocol.RES_VOID:
+			if (voidRes == null){
```

Comment:
How can any of these be null other than someone passed null to the methods above?
  Could that happen?